Can't cross-compile OTP/Erlang 25.3

**Describe the bug**
Tried to cross-compile OTP/Erlang 25.3 but make returns an error.


**To Reproduce**
$ ./configure --enable-bootstrap-only
$ make
$ ./configure --build=x86_64-linux-gnu --host=arm-linux-gnueabi --disable-dynamic-ssl-lib --with-ssl=/var/lib/cross/targets/arm/sysroot/usr erl_xcomp_sysroot=/var/lib/cross/targets/arm/sysroot/usr --disable-smp-require-native-atomics
$ make

And I got an error:
In file included from beam/erl_process.h:2924,
                 from beam/beam_code.h:25,
                 from beam/module.h:25,
                 from beam/global.h:34,
                 from sys/unix/erl_main.c:25:
beam/erl_process_lock.h: In function 'erts_proc_lock_wait_until_released':
beam/erl_process_lock.h:994: error: 'pix_lck' undeclared (first use in this function)
beam/erl_process_lock.h:994: error: (Each undeclared identifier is reported only once
beam/erl_process_lock.h:994: error: for each function it appears in.)

**Affected versions**
OTP 25.3

**Additional context**
System info:
Linux 2.6.31.8
armv5tel
arm-linux-gnueabi
gcc-4.2.2

I've been already succesfully using OTP 20.0 on my ARM.
And now I've decided to update  OTP.
I've succesfully built OTP 24.0 without smp-support and hipe but I can't build OTP 25.3
I used option:  --disable-smp-require-native-atomics
Because without this option I can't even configure project.
My compiler gcc-4.2.2 hasn't `gcc` `__atomic_*` builtins.  
And during configuration `gcc`  `__sync_*` builtins also wasn't found.
I tried to build with libatomic_ops but I couldn't configure project.
I got an error: ' no compare and swap'
So I configured with option: --disable-smp-require-native-atomics
Earlier I succesfully built OTP 20.0 with cross-compiler 4.9.4 with smp-support but I couldn't start it on my ARM.
I got message:
"A newer kernel is required to run this binary. (__kernel_cmpxchg64 helper)"
 So it seems my ARM doesn't support this function.

Last information:
This problem begins from OTP 25.0
OTP 24.3.4.10 is built successsfully

> And I got an error:
> In file included from beam/erl_process.h:2924,
> from beam/beam_code.h:25,
> from beam/module.h:25,
> from beam/global.h:34,
> from sys/unix/erl_main.c:25:
> beam/erl_process_lock.h: In function 'erts_proc_lock_wait_until_released':
> beam/erl_process_lock.h:994: error: 'pix_lck' undeclared (first use in this function)
> beam/erl_process_lock.h:994: error: (Each undeclared identifier is reported only once
> beam/erl_process_lock.h:994: error: for each function it appears in.)

I'm able to build without atomics, so I'm not able to reproduce this.

However, you *really*  do *not* want to do this. Performance will be horrible. We also do not have any test machines without native atomics support, so it might be/become buggy without us noticing.

> I tried to build with libatomic_ops but I couldn't configure project.

If you are building from the git repository, you need to run `autogen.sh` in the repository in order to get the `configure` script. Have you done that?

> Earlier I succesfully built OTP 20.0 with cross-compiler 4.9.4 with smp-support but I couldn't start it on my ARM.
> I got message:
> "A newer kernel is required to run this binary. (__kernel_cmpxchg64 helper)"
> So it seems my ARM doesn't support this function.

The cross tool-chain and configuration you're using must match the target OS and hardware. If not, you'll end up with issues like this.

I'd recommend upgrading both compiler and linux used since they are ancient. If not, you should at least use `libatomic_ops` for native atomics.

--- a/erts/emulator/beam/erl_process_lock.h
+++ b/erts/emulator/beam/erl_process_lock.h
@@ -991,6 +991,7 @@ erts_proc_lock_wait_until_released(Process *p, ErtsProcLocks locks)
 #if ERTS_PROC_LOCK_OWN_IMPL
 #if !ERTS_PROC_LOCK_ATOMIC_IMPL
         Uint32 was_locked;
+        erts_pix_lock_t *pix_lck = ERTS_PID2PIXLOCK(p->common.id);
 	erts_pix_lock(pix_lck);
 	was_locked =  (ERTS_PROC_LOCK_FLGS_READ_(&p->lock) & locks);
         erts_pix_unlock(pix_lck);

